### PR TITLE
DM-20548: Update HSC's camera geometry to use units of millimeters

### PR DESCRIPTION
**Pull request by laurenam**

### CODE REVIEW

File: python/lsst/fgcmcal/fgcmFitCycle.py [hunk 1]
```diff
@@ -978,34 +978,30 @@ def _loadCcdOffsets(self, butler):
...
             # this requires a pixelScale
             # Note that this now works properly with HSC, but I need to work on
             # generalizing this properly.  I expect the updates in DM-16490 will
             # generalize these computations, and appropriate tests can be added
             # on that ticket.
-            ccdOffsets['DELTA_RA'][i] = -pointXform.getY() * self.config.pixelScale / 3600.0
-            ccdOffsets['DELTA_DEC'][i] = -pointXform.getX() * self.config.pixelScale / 3600.0
-
+            ccdOffsets['DELTA_RA'][i] = -camPoint.getY() * self.config.pixelScale / 3600.0
+            ccdOffsets['DELTA_DEC'][i] = -camPoint.getX() * self.config.pixelScale / 3600.0
```
Comment (erykoff) [inline python/lsst/fgcmcal/fgcmFitCycle.py]: What is the value of `detector.getPixelSize()[0]`?  Is this not a drop-in replacement for `self.config.pixelScale`, which was an ugly hack to deal with the fact that the units of HSC were incorrect ... so I think all uses of `self.config.pixelScale` could be dropped.

Comment (laurenam) [reply python/lsst/fgcmcal/fgcmFitCycle.py]: May I refer you back to our discussion on Slack circa July 15 where you asked essentially the same question:  
"And if I’m not mistaken, the code is now scaling by the pixelSize (what does that return on the new version?  It’s the size in arcsec, right?) and unscaling by self.config.pixelScale (also the size in arcsec).  So we can remove both and it should work."

To which I replied:
"Sadly, no...pixelSize is in mm, not arcsec (obtained per detector from camera.py, e.g.: https://github.com/lsst/obs_subaru/blob/tickets/DM-20548/hsc/camera/camera.py#L76).  The closest I got to getting that seemingly elusive arcsec/pixel scaling was to compute the distance between the fieldAngle of two adjacent pixels computed via a PIXEL to FIELD_ANGLE transform.  However, this gives back an angular distance between pixels that includes the effects of the optical distortion model defined in the obs package (so it is slightly different for any given set of adjacent pixels — and the delta in the x vs. y position will differ slightly — depending where they lie in the FP) and this didn’t quite map to what you were computing."

Comment (laurenam) [reply python/lsst/fgcmcal/fgcmFitCycle.py]: And you had a final comment of:
"So then I think the plan of attack is that you do this simple shim right now.  And then when JohnP closes DM-20154, which I think will simplify the complicated dance, I’ll do the work from DM-16490
https://jira.lsstcorp.org/browse/DM-16490
(On my list, but blocked by this pixel scale).  So this isn’t work you need to/should do.  But it won’t be hard, and you have found something simple that works now and keeps things working."

Comment (erykoff) [reply python/lsst/fgcmcal/fgcmFitCycle.py]: :rofl: 
Okay then!  Thanks for remembering.

**Pull request merged**